app/models/roster/creator.rb

```diff
@@ -72,13 +75,15 @@ def perform
     rescue Result::Error, ActiveRecord::ActiveRecordError => err
       Result.failed(@roster, err.message)
     end
+    # rubocop:enable Metrics/MethodLength
 
     private
 
-    def add_identifiers_to_roster(raw_identifiers_string)
+    def add_identifiers_to_roster(raw_identifiers_string, google_ids: [])
       identifiers = raw_identifiers_string.split("\r\n").reject(&:blank?).uniq
-      identifiers.each do |identifier|
-        @roster.roster_entries << RosterEntry.new(identifier: identifier)
+
+      identifiers.zip(google_ids).each do |identifier, google_user_id|
```

Comment:
If there are no `google_ids` this just puts a nil for each roster entry

Reply:
I believe so. If there is an existing roster and we import Google Classroom, there will be more roster entries than the google_ids.

Reply:
Just talked to @srinjoym and there would not be a chance that these are uneven. We are grabbing the google classroom identifiers and ids to the roster entries.